### DISCUSSION TITLE
how to get the customer‘s carts ？ if I don't have the cart_id

### DISCUSSION THREAD

**Discussion by TDLiShiXiao** [Shopify/storefront-api-feedback#232] [Help]
we need implement that get the customer’s cart when the customer's call is coming ，we only have the customer id

**Comment by vixdug** [comment]
Hi @TDLiShiXiao It is not possible today to query a cart by anything other than a Cart ID. We are working on a feature to associate customers with carts and be able to query an authenticated customer for their carts. I don't have a timeline but I expect this would be released around 2024-07

**Comment by vysaghkt** [comment]
Hi @vixdug, Is it available now or are you still working on it? I can't see any queries to fetch the cart Id that is associated with a customer.
